pallets/chain-space/src/tests.rs

```diff
@@ -1082,3 +1082,48 @@ fn restoring_approval_of_a_space_by_non_root_should_fail() {
 		);
 	});
 }
+
+#[test]
+fn add_delegate_should_fail_if_space_delegates_limit_exceeded() {
+	let creator = DID_00;
+	let author = ACCOUNT_00;
+	let space = [2u8; 256].to_vec();
+	let space_digest = <Test as frame_system::Config>::Hashing::hash(&space.encode()[..]);
+
+	let id_digest = <Test as frame_system::Config>::Hashing::hash(
+		&[&space_digest.encode()[..], &creator.encode()[..]].concat()[..],
+	);
+
+	let space_id: SpaceIdOf = generate_space_id::<Test>(&id_digest);
+
+	new_test_ext().execute_with(|| {
+		// Create the space
+		assert_ok!(Space::create(
+			DoubleOrigin(author.clone(), creator.clone()).into(),
+			space_digest,
+		));
+
+		// Add the maximum number of delegates to the space
+		let mut delegate_count: u8 = 0;
+		while let Ok(_) = Space::space_delegate_addition(
```

Comment:
Since `MaxSpaceDelegates` is set at 10_000 in runtime, this loop might run that many times.
It is ok if there is no other way to change just for tests as in mock.

Reply:
https://github.com/dhiway/cord/blob/develop/pallets/chain-space/src/mock.rs#L85
It is set to 5.

Reply:
I could add 4 delegates after which I got an error (creator was the 1st delegate ig hence there was space for 4 more), so I have modified the code to for loop 4 times to fill the space.